### DISCUSSION TITLE
Add ability to edit and commit spreadsheets directly in GitHub

### DISCUSSION THREAD

**Discussion by lancejpollard** [community/community#54603] [Repositories]
### Select Topic Area

Product Feedback

### Body

[Here](https://github.com/tunebond/bead/blob/a63b154a19352a8fc52cf174c58ae7838bf83428/term.csv) is a sample CSV file which looks great when rendered. But when you click to edit, it takes you to the raw CSV file:

---

[![2023-05-05 22 21 38](https://user-images.githubusercontent.com/28718/236601835-166734c3-1b87-4165-bd62-23275a838bcb.gif)](https://github.com/tunebond/bead/blob/a63b154a19352a8fc52cf174c58ae7838bf83428/term.csv)

---

It would be **AMAZING** if one could edit the spreadsheet directly in either Google Sheets (syncing between the two somehow), or directly in GitHub in a simplified spreadsheet editing experience.

If a simplified GitHub spreadsheet experience is the way to go, then you would maybe have it so that anyone with a non-code ability could edit directly in GitHub. I have a lot of projects which could benefit from users editing the CSV data directly in GitHub, rather than having me create an XLSX from my GitHub CSV, having them edit that, then having me write a Node.js script to sync it back to the original CSV file (which only really works if one person edits the CSV at a time, which might take them several days to finish editing it...). Ideally I could just point them to a GitHub link and off they could go to editing the CSV in a spreadsheet-like experience!

Any ideas if you could have this as a feature at some point in the future? If not on the roadmap or not possible to add, then what do you recommend for simplifying this old-school workflow (which reminds me of emailing code around before there was svn/git haha)?

Initial search on solutions led to https://stackoverflow.com/questions/75495044/read-a-csv-from-github-to-google-sheets, which isn't too much to go off of.

**Comment by ntrip6** [comment]
Since MS owns Github we would more likely see csv editor point to Excel Online or to some open variant.

**Comment by johnyaku** [comment]
A simple way of getting half way there would be make the `Preview` button file-type aware.
GitHub is clearly capable of rendering CSV files as tables, so if I am editing a file called `file.csv` then it would be great if the preview could render it in the table -- in the same way as it will be rendered once the file is actually committed.

**Comment by RonNefyodovZen** [comment]
Totally agree, will be helpful

**Comment by u00w** [comment]
gets a vote from me!
